include/translatable-object.php

```diff
@@ -371,15 +373,18 @@ public function get_objects_with_no_lang( $limit, array $args = array() ) {
 			return array();
 		}
 
+		if ( ! $this instanceof PLL_Translated_Object ) {
+			return array();
+		}
+
 		$key          = md5( $sql );
-		$cache_type   = "{$this->type}s";
-		$last_changed = wp_cache_get_last_changed( $cache_type );
-		$cache_key    = "{$cache_type}_no_lang:{$key}:{$last_changed}";
-		$object_ids   = wp_cache_get( $cache_key, $cache_type );
+		$last_changed = wp_cache_get_last_changed( $this->cache_type );
+		$cache_key    = "{$this->cache_type}_no_lang:{$key}:{$last_changed}";
```

Comment:
I suggest to create a dedicated method to retrieve the cache key, (like it's done for WooCommerce if I say correctly). This way anyone will be able to get the key easily and invalidate the cache anytime.